src/GrantType/GrantTypeBase.php

```diff
@@ -86,7 +87,14 @@ public function getToken()
             $requestOptions = array_merge_recursive($requestOptions, $additionalOptions);
         }
 
-        $response = $this->client->post($config['token_url'], $requestOptions);
+        try {
+            $response = $this->client->post($config['token_url'], $requestOptions);
+        } catch (ClientException $e) {
+            if ($e->getResponse()->getStatusCode() === 400)  {
+                throw new RefreshTokenExpired();
+            }
```

Comment:
As we're now catching `GuzzleHttp\Exception\ClientException`, do we want to rethrow this if `if ($e->getResponse()->getStatusCode() === 400)` is false?
So
```
if ($e->getResponse()->getStatusCode() === 400)  {
    throw new RefreshTokenExpired();
} else {
    throw $e;
}
```